.circleci/config.yml

```diff
@@ -70,13 +70,33 @@ jobs:
 
   test-core:
     resource_class: xlarge
+    environment: # environment variables for the build itself
+      TEST_RESULTS: /tmp/test-results # path to where test results will be saved
     <<: *plain-go114
 
     steps:
       - checkout
+      - run: mkdir -p $TEST_RESULTS
+      - restore_cache: # restores saved cache if no changes are detected since last run
+          keys:
+            - go-mod-v4-{{ checksum "go.sum" }}
```

Comment:
The extra `restore_cache` and `save_cache` are documented in the [Go language guide](https://circleci.com/docs/2.0/language-go/). It's not clear if they are needed for capturing the test results, it seems to work without, but in theory it might speed up our tests a little.

I'm also not sure if it's working correctly, given the presence of [this in the CI logs](https://app.circleci.com/pipelines/github/DataDog/dd-trace-go/1510/workflows/f9d53026-6371-4003-8a7f-14dea37c877e/jobs/10270). So we could also omit it from this PR for now as far as I'm concerned.

<img width="1501" alt="2021-06-28 Google Chrome test-core (10270) - DataDogdd-trace-go - 19-15-14" src="https://user-images.githubusercontent.com/15000/123677265-2fba1780-d845-11eb-9f5c-df774d72a452.png">